### PR TITLE
feat: Add ECMAScript 6 (ES6) support

### PR DESCRIPTION
**Pull request by dplewis**
### New Pull Request Checklist

- [ ] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server-push-adapter/security/policy).
- [ ] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server-push-adapter/issues?q=is%3Aissue).

### Issue Description

The code base doesn't support async/await since babel didn't support it in the runtimegenerator. We could update babel but I don't think project needs to be trancompiled. The test suite ran against `/src` instead of `/lib` so this error didn't get caught

```
*/node_modules/@parse/push-adapter/lib/WEB.js:59
var _ref = _asyncToGenerator( /#PURE/regeneratorRuntime.mark(function _callee(data, devices) {
^
```
Closes: https://github.com/parse-community/parse-server-push-adapter/issues/256

### Approach

* Add `"type": "module"` to `package.json` to enable ES6
* Remove babel transcompiler
* Remove unused files
* Randomize test suite
* Use `C8` for test coverage
* Bump testing dependencies
* Improve CI npm cache

### TODOs before merging

- [x] Add tests
- [x] Add entry to changelog
- [x] Add changes to documentation (guides, repository pages, in-code descriptions)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,14 +37,10 @@
...
     "@semantic-release/release-notes-generator": "9.0.3",
-    "babel-cli": "6.26.0",
-    "babel-core": "6.26.3",
-    "babel-preset-es2015": "6.24.1",
-    "babel-preset-stage-0": "6.24.1",
-    "codecov": "3.7.1",
-    "jasmine": "2.8.0",
-    "jasmine-spec-reporter": "4.0.0",
-    "nyc": "14.1.1",
+    "c8": "10.1.2",
```
Comment (mtrezza) [inline package.json:40]: Why switch from nyc to c8? I believe we use nyc across our repos?

Comment (dplewis) [reply package.json:40]: `nyc` doesn't support es6 modules. Also c8 uses the same istanbul-reporter and `.nycrc` file as `nyc`.

**Review by mtrezza — APPROVED** [review]

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.3.0](https://github.com/parse-community/parse-server-push-adapter/releases/tag/6.3.0)

**Comment by Moumouls** [comment]
it seems that jest just exit if he detect ESM in commonjs, so the error detection don't work

**Comment by mtrezza** [comment]
Are there any issues with this? It's a rather old release and I'm unaware of any issues reported regarding that.

**Comment by kennic** [comment]
Why the hell on earth has this package been converted to ES6 while Parse-Server is still using CommonJS?
CONVERT IT BACK!!!!!

**Pull request merged**